### PR TITLE
Fixing the `stop_models` functions by calling explicitly the removal of the replica sets and pods

### PR DESCRIPTION
**Pull request by jacekwachowiak**
I have noticed that by calling the `stop_models` or `stop_all_model_containers` only a Kubernetes deployment is being removed, while the replica sets and pods remain. Even worse is that the pods are orphaned and cannot be managed or detected by Clipper. 
I added 2 calls to remove the corresponding replica sets and pods. From what I understand, using the Python API stops the Kubernetes calls from cascading, which means that the deployment may not remove the replica set and in effect the pods. I checked that even with a replica set deletion the pods are not gone, so to clean completely both remaining resources are removed with separate calls.
Issue #729

Some console output to confirm that it works
```
clipper_conn.get_all_models()

['stop-test:1', 'sum-model:1']

[cloud-user@node1 ~]$ kubectl get rs | grep deployment
stop-test-1-deployment-at-0-at-default-cluster-7b586b67f7   1         1         1       78s
sum-model-1-deployment-at-0-at-default-cluster-6c4f5bbdc7   1         1         1       92s

[cloud-user@node1 ~]$ kubectl get po | grep deployment
stop-test-1-deployment-at-0-at-default-cluster-7b586b67f7-tnvcp   1/1     Running   0          93s
sum-model-1-deployment-at-0-at-default-cluster-6c4f5bbdc7-fqtpx   1/1     Running   0          107s

clipper_conn.stop_models('sum-model')
19-06-13:11:54:01 INFO     [clipper_admin.py:1285] Model sum-model:1 was successfully deleted
19-06-13:11:54:01 INFO     [clipper_admin.py:1319] [default-default-cluster] Stopped all containers for these models and versions:
{'sum-model': ['1']}

[cloud-user@node1 ~]$ kubectl get rs | grep deployment
stop-test-1-deployment-at-0-at-default-cluster-7b586b67f7   1         1         1       2m8s
[cloud-user@node1 ~]$ kubectl get po | grep deployment
stop-test-1-deployment-at-0-at-default-cluster-7b586b67f7-tnvcp   1/1     Running       0          2m9s
sum-model-1-deployment-at-0-at-default-cluster-6c4f5bbdc7-fqtpx   1/1     Terminating   0          2m23s

clipper_conn.stop_all_model_containers()
19-06-13:11:54:54 INFO     [clipper_admin.py:1285] Model stop-test:1 was successfully deleted
19-06-13:11:54:54 INFO     [clipper_admin.py:1410] [default-default-cluster] Stopped all Clipper model containers:
{'stop-test': ['1']}

[cloud-user@node1 ~]$ kubectl get rs | grep deployment
[cloud-user@node1 ~]$ kubectl get po | grep deployment
stop-test-1-deployment-at-0-at-default-cluster-7b586b67f7-tnvcp   1/1     Terminating   0          2m58s
[cloud-user@node1 ~]$ kubectl get po | grep deployment
```

### CODE REVIEW

**Comment by simon-mo** [comment]
jenkins ok to test

**Review by simon-mo — APPROVED** [review]
@jacekwachowiak It's strange to see this kind of behavior in k8s api. but we have to live with it. Thanks for catching it

**Comment by simon-mo** [comment]
I'll wait for @withsmilo to review this before merging https://github.com/ucbrise/clipper/issues/729#issuecomment-501526343

File: clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py
Comment (withsmilo) [inline clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: It seems there are redundant codes. Would like to define `label_selector` as a variable?

File: clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py
Comment (withsmilo) [inline clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: It seems there are redundant codes. Would like to define `label_selector` as a variable?

**Comment by jacekwachowiak** [comment]
@simon-mo As I discussed with @withsmilo in the issue #729, there is a cleaner way to do  this with `propagation_policy='Foreground'` optional argument using `delete_namespaced_deployment()` instead of `delete_collection_namespaced_deployment()`, but this applies only to a single deployment and I cannot find a way to get all the models while calling the `stop_all_model_containers`. And I don't think it's a good idea to have two different implementations for the single model stopping and for all of them, so unless there is a way to get the list of all models inside the `stop_all_model_containers`, I see no other way

File: clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py
Comment (jacekwachowiak) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: Sure, I copy-pasted the parts so it's not pretty, I will fix it now :)

File: clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py
Comment (jacekwachowiak) [reply clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py]: Done

**Review by withsmilo — APPROVED** [review]
LGTM!

**Comment by rkooo567** [comment]
LGTM too! Let's merge it when test is passed.

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**
